src/bun.js/bindings/BunProcess.cpp

```diff
@@ -522,58 +536,101 @@ extern "C" void Process__dispatchOnExit(Zig::GlobalObject* globalObject, uint8_t
+
+    setProcessExitCodeInner(globalObject, process, code);
+    RETURN_IF_EXCEPTION(throwScope, {});
+
+    auto exitCode = Bun__getExitCode(bunVM(zigGlobal));
     Process__dispatchOnExit(zigGlobal, exitCode);
-    Bun__Process__exit(zigGlobal, exitCode);
+
+    // process.reallyExit(exitCode);
+    auto reallyExitVal = process->get(globalObject, Identifier::fromString(vm, "reallyExit"_s));
+    RETURN_IF_EXCEPTION(throwScope, {});
+    MarkedArgumentBuffer args;
+    args.append(jsNumber(exitCode));
+    JSC::call(globalObject, reallyExitVal, args, ""_s);
+    RETURN_IF_EXCEPTION(throwScope, {});
+
     return JSC::JSValue::encode(jsUndefined());
 }
 
-JSC_DEFINE_HOST_FUNCTION(Process_setUncaughtExceptionCaptureCallback,
-    (JSC::JSGlobalObject * globalObject, JSC::CallFrame* callFrame))
+JSC_DEFINE_HOST_FUNCTION(Process_getBuiltinModule, (JSC::JSGlobalObject * lexicalGlobalObject, JSC::CallFrame* callFrame))
 {
-    auto throwScope = DECLARE_THROW_SCOPE(globalObject->vm());
-    JSValue arg0 = callFrame->argument(0);
-    if (!arg0.isCallable() && !arg0.isNull()) {
-        throwTypeError(globalObject, throwScope, "The \"callback\" argument must be callable or null"_s);
-        return {};
+    auto* globalObject = reinterpret_cast<Zig::GlobalObject*>(lexicalGlobalObject);
+    auto& vm = globalObject->vm();
+    auto throwScope = DECLARE_THROW_SCOPE(vm);
+
+    auto id_value = callFrame->argument(0);
+    Bun::V::validateString(throwScope, globalObject, id_value, "id"_s);
+    RETURN_IF_EXCEPTION(throwScope, {});
+
+    auto id_str = id_value.toWTFString(globalObject);
+    RETURN_IF_EXCEPTION(throwScope, {});
+
+    auto is_valid_bultin_name = Bun::isBuiltinModule(id_str);
+    if (!is_valid_bultin_name) return JSValue::encode(jsUndefined());
+
+    auto require_key = id_value.toString(globalObject);
+
+    JSCommonJSModule* moduleObject = nullptr;
+
+    auto entry = globalObject->requireMap()->get(globalObject, id_value);
+    if (!entry.isUndefined()) moduleObject = jsDynamicCast<JSCommonJSModule*>(entry);
+
+    if (!moduleObject) {
+        moduleObject = Bun::JSCommonJSModule::create(vm, globalObject->CommonJSModuleObjectStructure(), require_key, require_key, jsEmptyString(vm), SourceCode());
+        // TODO: if this is uncommented and `process.getBuiltinModule` runs before `require` then the function here returns `{}`
+        // but CommonJSModuleRecord.cpp calls it so it seems likely to be necessary, otoh the tests work without it /shrug
+        // globalObject->requireMap()->set(globalObject, require_key, moduleObject);
     }
-    auto* zigGlobal = defaultGlobalObject(globalObject);
-    jsCast<Process*>(zigGlobal->processObject())->setUncaughtExceptionCaptureCallback(arg0);
+
+    auto fn = globalObject->requireFunctionUnbound();
+    MarkedArgumentBuffer args;
+    args.append(id_value);
+    return JSValue::encode(JSC::profiledCall(globalObject, ProfilingReason::API, fn, JSC::getCallData(fn), moduleObject, args));
```

Comment:
Calling require on it is unnecessary work, and also means that the user can override the exports.